File: Exercism/c/triangle/triangle.c
#include "triangle.h"

static bool is_triangle(triangle_t sides);

bool is_equilateral(triangle_t sides)
{
    bool isEquilateral = false;

    if(is_triangle(sides))
    {
        if(sides.a == sides.b && sides.a == sides.c && sides.a == sides.c)
        {
            isEquilateral = true;
        }
    }

    return isEquilateral;
}

bool is_isosceles(triangle_t sides)
{
    bool isIsosceles = false;

    if(is_triangle(sides))
    {
        if( sides.a == sides.b ||
            sides.a == sides.c ||
            sides.b == sides.c)
        {
            isIsosceles = true;
        }
    }

    return isIsosceles;
}

bool is_scalene(triangle_t sides)
{
    bool isScalene = false;
    
    if(is_triangle(sides))
    {
        if( sides.a != sides.b && 
            sides.a != sides.c &&
            sides.b != sides.c)
        {
            isScalene = true;
        }
    }

    return isScalene;
}

static bool is_triangle(triangle_t sides)
{
    bool isTriangle = false;

    if(sides.a != 0 && sides.b != 0 && sides.c != 0)
    {
        if( sides.a + sides.b >= sides.c &&
            sides.b + sides.c >= sides.a &&
            sides.a + sides.c >= sides.b)
        {
            isTriangle = true;
        }
    }

    return isTriangle;
}

File: Exercism/python/triangle/triangle.py
def equilateral(sides):

    isEquilateral = False

    if is_triangle(sides):
        if (sides[0] == sides[1]) and (sides[0] == sides[2]) and (sides[1] == sides[2]):
            isEquilateral = True
    
    return isEquilateral

def isosceles(sides):

    isIsosceles = False

    if is_triangle(sides):
        if sides[0] == sides[1] or sides[0] == sides[2] or sides[1] == sides[2]:
            isIsosceles = True

    return isIsosceles


def scalene(sides):
    isScalene = False

    if is_triangle(sides):
        if (sides[0] != sides[1]) and (sides[0] != sides[2]) and (sides[1] != sides[2]):
            isScalene = True

    return isScalene

def is_triangle(sides):
    isTriangle = False

    if (sides[0] != 0) and (sides[1] != 0) and (sides[2] != 0):
        if (sides[0] + sides[1] >= sides[2]) and (sides[1] + sides[2] >= sides[0]) and (sides[0] + sides[2] >= sides[1]):
            isTriangle = True
    
    return isTriangle

